src/Compilers/CSharp/Portable/Binder/Binder_Query.cs

```diff
@@ -797,6 +820,24 @@ protected BoundCall MakeQueryInvocation(CSharpSyntaxNode node, BoundExpression r
                 // Queries are syntactical rewrites, so we allow fields and properties of delegate types to be invoked,
                 // although no well-known non-generic query method is used atm.
                 allowFieldsAndProperties: true);
+
+            static BoundExpression updateUltimateReceiver(BoundExpression receiver, BoundExpression originalUltimateReceiver, BoundExpression replacementUltimateReceiver)
+            {
+                if (receiver is BoundQueryClause query)
+                {
+                    return query.Update(
```

Comment:
>return query.Update( [](start = 20, length = 20)

Is this code path covered by an y test?

Reply:
Yes. Both indexed property cases hit this: they receiver is wrapped in a query clause unconditionally on line 64 in this file.

---
In reply to: [556915303](https://github.com/dotnet/roslyn/pull/50322#discussion_r556915303) [](ancestors = 556915303)